extension/content/components/recipes/form/arguments/OptOutStudy.js

```diff
@@ -0,0 +1,67 @@
+import React from "react";
+import { Grid, Row, Col } from "rsuite";
+
+import { useRecipeDetailsData } from "devtools/contexts/recipeDetails";
+import { useSelectedNormandyEnvironmentAPI } from "devtools/contexts/environment";
+import InputField from "devtools/components/recipes/form/arguments/fields/InputField";
+import SelectField from "devtools/components/recipes/form/arguments/fields/SelectField";
+import ToggleField from "devtools/components/recipes/form/arguments/fields/ToggleField";
+
+export default function OptOutStudy() {
+  const [extensions, setExtensions] = React.useState([]);
+  const normandyApi = useSelectedNormandyEnvironmentAPI();
+
+  React.useEffect(() => {
+    normandyApi.fetchAllExtensions().then((allExtensions) => {
+      setExtensions(allExtensions);
+    });
+  }, []);
```

Comment:
See: https://github.com/mozilla-extensions/normandy-devtools/pull/113/files#diff-436b1db721b074d7c034f080f51a31d7R16-R27

You need this effect applied any time the environment key changes.